pkg/monitor/monitor/util.go

```diff
@@ -1003,3 +1032,124 @@ func getMonitorVolumeClaims(monitor *v1alpha1.TidbMonitor) []core.PersistentVolu
 	}
 	return nil
 }
+
+func getThanosSidecarContainer(monitor *v1alpha1.TidbMonitor) core.Container {
+	bindAddress := "[$(POD_IP)]"
+	if monitor.Spec.Thanos.ListenLocal {
```

Comment:
What about defining a variable `thanos:=monitor.Spec.Thanos` to simplify the code.